src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java

```diff
@@ -121,9 +128,28 @@ public void addUserToGroup(String user, String group) throws IOException {
     }
 
     @Override
-    protected void setPermissions(FileSystemPermission permission, Path path,
-                                  EnumSet<FileSystemPermission.Option> options) throws IOException {
-        // [P41372857]: Implement using ACL for Windows
+    protected void setOwner(FileSystemPermission permission, Path path) throws IOException {
+        UserPrincipalLookupService lookupService = path.getFileSystem().getUserPrincipalLookupService();
+        UserPrincipal userPrincipal = lookupService.lookupPrincipalByName(permission.getOwnerUser());
+        FileOwnerAttributeView view = Files.getFileAttributeView(path, FileOwnerAttributeView.class,
+                LinkOption.NOFOLLOW_LINKS);
+
+        logger.atTrace().setEventType(SET_PERMISSIONS_EVENT).kv(PATH, path).kv("owner", permission.getOwnerUser())
+                .log();
+        view.setOwner(userPrincipal);
+
+        // Note that group ownership is not used.
```

Comment:
I don't see a file can be owned by a group in Windows.

Reply:
it's not really about ownership, but permissions. The group should have some permissions on the file/folder

Reply:
The problem is which group? There is no default group AFAIK.

Reply:
Sorry, I was confused with `groupRead`, etc which applies to the default group and there is no default group in Windows. 

Back to your feedback: Would you recommend using the passed in group and allow such group to have some basic read/write permission?

Reply:
No, it should have the group permissions that are being requested through setPermission. If the API isn't working for you, can you refactor it so that it does work? 

I believe the api used to just simply be setPermission which took a permission, user, and group.

Reply:
Make sense. I think I can add the permission accordingly [here](https://github.com/aws-greengrass/aws-greengrass-nucleus/pull/903/files#diff-1848e4b7f214f13eb0232bdb3c0d8de3b38e7e84b88b690941d7c4c991392d7fR233).

Reply:
Updated. Please review. Thanks for pointing this out!